packages/application/src/connectionlost.ts

```diff
@@ -22,6 +25,45 @@ export const ConnectionLost: IConnectionLost = async function (
       'JupyterLab will continue trying to reconnect.\n' +
       'Check your network connection or Jupyter server configuration.\n'
   );
+  const buttons = [Dialog.okButton({ label: trans.__('Dismiss') })];
+
+  if (!displayConnectionLost) {
+    return;
+  }
+
+  const key = 'server-connection-error';
+  const promise = connectionLostCachePromise[key];
```

Comment:
After the recent changes, I do not see a reason for this to be a mapping if the key is constant.